### PR TITLE
Syntactic fixes (edit: and some minor semantic fixes)

### PR DESCRIPTION
**Pull request by RadioPotin**
Hello,
Fixed some syntactic issues with some of your `.c` files.
Cordially

### CODE REVIEW

**Comment by RadioPotin** [comment]
Hello again,

This message is to bring you some context for the merge request I've opened, particularly on the semantic changes I brought to your code.

I am currently reviewing several C benchmarks with Frama-C in the context of my internship.

When running Frama-C on some of the tests in your benchmark, we have identified some situations which seem not to correspond to the intended meaning of the test cases (i.e. some "good"\* test cases with undefined behaviour, and some "bad"\*\* test cases with unrelated issues), at least if a thorough use of Frama-C is to be conducted.

We have applied a few fixes which we believe improve the test cases while keeping the intended behaviour.

Know that we have thoroughly verified that our modifications did not change the results of your original work, and have modified only the cases that we know should be handled by Frama-C, thus leaving unsupported features like `recursion` or `file_access` test cases untouched.

\**"good" cases correspond to executions in which no defect should be detected by the analyzer but were due to unrelated issues.*
\*\**"bad" cases correspond to executions in which the defect were supposed to be detected by the analyzer but weren't.*

These faulty cases are found in the following files:

## - memory/double\_free/10\_loop\_while\_continue.c (good case)

In this file, the block that contains the dangerous instruction was moved a few lines up so that:

When the parameter `x` is equal to `9`, as the bad case execution requests, the conditional is met and the double call to `free` is reached, contrary to the situation in which `x` is equal to `10`(the case in which no error should be reached), thus not meeting the conditional and no faulty instruction is reached.

## - general/nullpointer/07\_cross\_file.c (good case)

This file has had a slight modification to the good case scenario.
Indeed, the instructions `val = arr[2];` and `val = ptr[2];` were offsetted by 1 so that, in the good case, the read was not out of bounds but remained so in the bad case when it was needed that it was.

## - integers/overflow/05\_simple\_goto.c (good case)

Here there are two possible overflows happening, one at instruction `printf("%i\n", val * 3);` and the 
other at `val = val * 2;`

This causes an unintended signed overflow, which is undefined behavior in the C standard; converting the numeric literal to unsigned with the u is sufficient to ensure the multiplication is performed on unsigned integers (whose overflow is well-defined in the standard), and then downcast back to the signed integer val (which is also well-defined).

Indeed it happened everytime as soon as the first call to the `GOTO_LABEL` was done since the instruction `val = val * 2;` was reached everytime. `u` allows to turn an integer overflow undefined behaviour into a unsigned integer overflow which is defined in the standard. (*"A computation involving unsigned operands can never overflow (...)"*)

## - integers/overflow/07\_simple\_cross\_file.c (bad case)

Here we have only added an `if () {...} else {}` statement to allow the overflow to happen as intended in the bad case. Function `choose_int(int, int, bool);` as it was called originally never returned a value capable of overflowing in the bad case.

## - integers/overflow/09\_loop\_for\_complex.c (good case)
## - integers/overflow/10\_loop\_while\_continue.c (good case)
## - integers/overflow/11\_loop\_while\_do\_continue.c (good case)

To these three files we have brought the same changes because the value at which variable `val` was initialized caused an overflow in every case.

Thank you for taking the time to read and consider this MR, I remain at your disposal if any further information is required.

Cordially,
Dario Pinto.

File: integers/overflow/07_simple_cross_file.c [hunk 1]
```diff
@@ -13,7 +13,12 @@ void integer_overflow_cross_file(bool a, bool b) {
     }
 #endif
 
-    val = choose_int(2, INT_MAX / 3 - 1, INT_MAX / 2 - 1);
+    if (!a && !b) {
+        val = choose_int(2, INT_MAX / 2 - 1, 0);
+    }
+    else
+        val = choose_int(2, INT_MAX / 3 - 1, INT_MAX / 2 - 1);
+
```
Comment (JMoerman) [inline integers/overflow/07_simple_cross_file.c:21]: I think that this is a better fix as I think that `a` should have been the third argument: 
```suggestion
    val = choose_int (INT_MAX / 3 - 1, INT_MAX / 2 - 1, a);
```
See https://github.com/JMoerman/JM2018TS/blob/ce53f08350648a878772f844e05ef5c56cb2d004/common/common.c#L37

File: integers/overflow/05_simple_goto.c [hunk 2]
```diff
@@ -21,7 +21,7 @@ void integer_overflow_goto(int x) {
         printf("%i\n", val);
     }
 
-    val = val * 2;
+    val = val * 2u;
 
     if(i == 0) {
         i++;
```
Comment (JMoerman) [inline integers/overflow/05_simple_goto.c:27]: I don't think that I meant for this line to cause any kind of overflow. It only needs to be executed once to cause an overflow in `printf("%i\n", val * 3);` so placing the line in the last if statement seems to be a better idea.
```suggestion
    if(i == 0) {
        i++;
        val = val * 2;
```

**Review by JMoerman — COMMENTED** [review]
Thanks for the MR! I feel a bit clumsy when looking at these mistakes 😅️.
Most changes look good, but I have some comments on your changes on `integers/overflow/07_simple_cross_file.c` and `integers/overflow/05_simple_goto.c`.

File: integers/overflow/07_simple_cross_file.c (hunk 1 above)
Comment (RadioPotin) [reply integers/overflow/07_simple_cross_file.c:21]: Yes I totally get it, it does seem better that way !

File: integers/overflow/05_simple_goto.c (hunk 2 above)
Comment (RadioPotin) [reply integers/overflow/05_simple_goto.c:27]: This does seem to do the trick !

**Comment by RadioPotin** [comment]
Happy to have been able to better the benchmark, even so slightly.

Again, thank you for your time and for your quality test suite, best regards.
Dario Pinto, intern in the FramaC team ! :)

**Comment by JMoerman** [comment]
I'm glad that the benchmark is of use to someone else!

**Pull request merged**